cmake/Options/TileDBAssertions.cmake

```diff
@@ -0,0 +1,29 @@
+# Lightly modified from LLVM, used under "Apache 2.0 License with LLVM exceptions”
+# https://github.com/llvm/llvm-project/blob/93d1a623cecb6f732db7900baf230a13e6ac6c6a/llvm/cmake/modules/HandleLLVMOptions.cmake#L60-L85
+
+if(TILEDB_ASSERTIONS)
+  # MSVC doesn't like _DEBUG on release builds. See LLVM PR 4379.
+  if( NOT MSVC )
+    add_definitions( -D_DEBUG )
+  endif()
+  # On non-Debug builds cmake automatically defines NDEBUG, so we
+  # explicitly undefine it:
+  if( NOT uppercase_CMAKE_BUILD_TYPE STREQUAL "DEBUG" )
```

Comment:
Is tiledb cmake infrastructure defining 'uppercase_CMAKE_BUILD_TYPE somewhere, or is that cmake magic?
(https://gitlab.kitware.com/cmake/cmake/-/issues/22591 makes it sound like it's clang specific...?)